engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowExecutionActor.scala

```diff
@@ -20,14 +28,58 @@ object WorkflowExecutionActor {
 
   case object WorkflowExecutionPendingState extends WorkflowExecutionActorState
   case object WorkflowExecutionInProgressState extends WorkflowExecutionActorState
-  case object WorkflowExecutionSuccessfulState extends WorkflowExecutionActorTerminalState
-  case object WorkflowExecutionFailedState extends WorkflowExecutionActorTerminalState
+  case object WorkflowExecutionSuccessfulState extends WorkflowExecutionActorTerminalState{
+    override val terminal = true
+  }
+  case object WorkflowExecutionFailedState extends WorkflowExecutionActorTerminalState {
+    override val terminal = true
+  }
   case object WorkflowExecutionAbortedState extends WorkflowExecutionActorTerminalState
 
+  case class SymbolCacheKey(scopeName: String, input: Boolean)
+
+  type SymbolCache = Map[SymbolCacheKey, Traversable[SymbolStoreEntry]]
+  type ExecutionStore = Map[JobKey, ExecutionStatus]
+  type ExecutionStoreEntry = (JobKey, ExecutionStatus)
+
   /**
     * State data
     */
-  final case class WorkflowExecutionActorData()
+  final case class WorkflowExecutionActorData(executionStore: ExecutionStore,
+                                              symbolCache: SymbolCache,
+                                              callOutputs: Map[JobKey, CallOutputs] = Map.empty) {
+
+    /** This method updates: The ExecutionStore with the updated status, the symbol cache with the new outputs, and appends to the overall CallOutputs */
+    def handleCompletedCall(jobKey: JobKey,
```

Comment:
To be pedantic, should be `Job` rather than `Call`.

I also wasn't too thrilled with `handle` since this isn't really a handler...